Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -305,14 +404,8 @@ private extension UInt8 {
         case .keepAlive:
             return true
         case .unknown:
-            for header in self.headers {
-                if self.buffer.equalCaseInsensitiveASCII(view: "connection".utf8, at: header.name) {
-                    if self.buffer.equalCaseInsensitiveASCII(view: "close".utf8, at: header.value) {
-                        return false
-                    }
-                    return self.buffer.equalCaseInsensitiveASCII(view: "keep-alive".utf8, at: header.value)
-                }
-            }
+            if let keepAliveStatus = self.isKeepAlive { return keepAliveStatus }
```

Comment:
No need to make this fit on one line, you can put the conditional body on the next line.